### ISSUE TITLE
`docker wait` hangs for non-existent containers

### ISSUE THREAD

**Issue by spiffxp** [docker/for-mac#1714]
### Expected behavior
```
$ docker wait does-not-exist
Error response from daemon: No such container: does-not-exist
```

### Actual behavior
```
$ docker wait does-not-exist
# ... hangs, must Ctrl-C out
```

### Information
  - Full output of the diagnostics from "Diagnose & Feedback" in the menu
```
Docker for Mac: version: 17.06.0-rc2-ce-mac14 (4cdec4294a50b2233146b09469b49937dabdebdd)
macOS: version 10.11.6 (build: 15G1510)
logs: /tmp/89D5EC98-56AB-4C11-9F39-A99D81773C09/20170609-093708.tar.gz
[OK]     db.git
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     db
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     slirp
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[OK]     docker-cli
[OK]     menubar
[OK]     disk
```

**Comment by samirault** [comment]
Also failing in the same way on linux:

```
Client:
 Version:      17.06.0-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:23:31 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.0-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   02c1d87
 Built:        Fri Jun 23 21:19:04 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

**Comment by ianchakeres** [comment]
I ran into this issue with the edge channel docker for mac while trying to build kubernetes, as it blocks the make quick-release build.

Posting here to help anyone else who runs into the same issue while trying to build kubernetes.

**Comment by ianchakeres** [comment]
The latest stable version Docker 17.06.0-ce-mac18 is now failing for me.

I rolled back to the prior stable version 17.03.1-ce-mac12 (17661).

**Comment by spiffxp** [comment]
Confirmed I'm seeing this with docker 17.06.0-ce-mac18.  

May be related to https://github.com/moby/moby/pull/32237 landing

**Comment by thaJeztah** [comment]
Thanks for reporting; this is a duplicate of https://github.com/moby/moby/issues/33948, and already resolve on master through https://github.com/moby/moby/pull/33293, which will be included in docker 17.06.1, that is currently being worked on.

**Comment by spiffxp** [comment]
This has been fixed
